### PR TITLE
DFOCVC stories

### PR DESCRIPTION
**Pull request by PhysicsJon**
[AB#11203](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/11203)
[AB#11202](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/11202)
[AB#11258](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/11258)

### CODE REVIEW

File: src/NHSDPublicBrowseAcceptanceTests.TestData/Utils/Queries.cs [hunk 1]
```diff
@@ -104,6 +104,12 @@ from Solution
         public const string DeleteCatalogueItem = "DELETE FROM CatalogueItem WHERE CatalogueItemId=@CatalogueItemId";
         public const string GetAllCatalogueItems = "SELECT * FROM [dbo].[CatalogueItem]";
 
+        public const string GetFrameworkSolutionCount = @"SELECT COUNT(*)
+                                                              FROM FrameworkSolutions AS fs
+                                                              INNER JOIN CatalogueItem AS ci ON ci.CatalogueItemId = fs.SolutionId
+                                                              WHERE FrameworkId IN(SELECT TOP(1) Id FROM Framework WHERE ShortName = @frameworkName)
+                                                              AND ci.PublishedStatusId IN (3, 4)";
+
```
Comment (bjss-chris) [inline src/NHSDPublicBrowseAcceptanceTests.TestData/Utils/Queries.cs]: Tend to recommend including the schema name when referencing tables, so `dbo.FrameworkSolutions` and `dbo.CatalogueItem`.

File: src/NHSDPublicBrowseAcceptanceTests.TestData/Utils/Queries.cs [hunk 2]
```diff
@@ -104,6 +104,12 @@ from Solution
...
         internal const string GetLastUpdated = "SELECT LastUpdated FROM @table WHERE @whereKey=@whereValue";
```
Comment (bjss-chris) [inline src/NHSDPublicBrowseAcceptanceTests.TestData/Utils/Queries.cs]: Know it's not changed as part of this PR but think operators are more readable surrounded by spaces (same on line 104).

File: src/NHSDPublicBrowseAcceptanceTests.Tests/NHSDPublicBrowseAcceptanceTests.Tests.csproj [hunk 3]
```diff
@@ -59,4 +67,10 @@
     </SpecFlowFeatureFiles>
   </ItemGroup>
 
+  <ItemGroup>
+    <Compile Update="Features\DFOCVC\ViewDFOCVCSolutionsList.feature.cs">
+      <DependentUpon>ViewDFOCVCSolutionsList.feature</DependentUpon>
+    </Compile>
+  </ItemGroup>
```
Comment (bjss-chris) [inline src/NHSDPublicBrowseAcceptanceTests.Tests/NHSDPublicBrowseAcceptanceTests.Tests.csproj]: Are all the additions to the project file definitely required? I think I've seen SpecFlow add superfluous entries to project files before.

File: src/NHSDPublicBrowseAcceptanceTests.TestData/Utils/Queries.cs [hunk 4]
```diff
@@ -76,34 +76,40 @@ from Solution
...
+        public const string GetAllCatalogueItems = "SELECT * FROM [dbo].[CatalogueItem];";
 
-        internal const string GetLastUpdated = "SELECT LastUpdated FROM @table WHERE @whereKey=@whereValue";
+        public const string GetFrameworkSolutionCount = @"SELECT COUNT(*)
+                                                              FROM FrameworkSolutions AS fs
+                                                              INNER JOIN CatalogueItem AS ci ON ci.CatalogueItemId = fs.SolutionId
+                                                              WHERE FrameworkId IN(SELECT TOP(1) Id FROM Framework WHERE ShortName = @frameworkName)
+                                                              AND ci.PublishedStatusId IN (3, 4);";
+
+        internal const string GetLastUpdated = "SELECT LastUpdated FROM @table WHERE @whereKey=@whereValue;";
```
Comment (bjss-chris) [inline src/NHSDPublicBrowseAcceptanceTests.TestData/Utils/Queries.cs:113]: Terminating semicolons – like it! 😃 

There are a few other improvements that you could make to the SQL formatting but non-essential, e.g.

1. Space after comma.
2. Square brackets only where required.
3. Use schema name when referencing tables.
4. Spaces around operators.

File: src/NHSDPublicBrowseAcceptanceTests.Tests/NHSDPublicBrowseAcceptanceTests.Tests.csproj (hunk 3 above)
Comment (PhysicsJon) [reply src/NHSDPublicBrowseAcceptanceTests.Tests/NHSDPublicBrowseAcceptanceTests.Tests.csproj]: I'll get those removed, never seem to figure out why they are added either

File: src/NHSDPublicBrowseAcceptanceTests.TestData/Utils/Queries.cs (hunk 4 above)
Comment (PhysicsJon) [reply src/NHSDPublicBrowseAcceptanceTests.TestData/Utils/Queries.cs:113]: I'll get around to those as and when I look back to reformatting these queries

**Review by bjss-chris — APPROVED** [review]

**Pull request merged**